File: frontend/src/types/document.ts
export type DocumentStatus =
  | "pending"
  | "extracting"
  | "chunking"
  | "embedding"
  | "indexed"
  | "error";

export interface Document {
  id: string;
  filename: string;
  file_format: string;
  upload_time: string;
  status: DocumentStatus;
  error_message: string | null;
  page_count: number | null;
  chunk_count: number | null;
}

export interface UploadResponse {
  document_id: string;
  filename: string;
  status: string;
}

export interface DocumentListResponse {
  documents: Document[];
}

File: backend/app/models/document.py
from datetime import datetime
from typing import Optional

from sqlmodel import Field, SQLModel


class Document(SQLModel, table=True):
    id: str = Field(primary_key=True)
    filename: str
    file_format: str  # pdf | epub | docx | doc | md | txt
    upload_time: datetime = Field(default_factory=datetime.utcnow)
    status: str = Field(default="pending")  # pending | extracting | chunking | embedding | indexed | error
    error_message: Optional[str] = None
    page_count: Optional[int] = None
    chunk_count: Optional[int] = None
    qdrant_ids: Optional[str] = None  # JSON array of Qdrant point UUIDs
